accepted/2020/net5/net5.md

```diff
@@ -397,6 +397,167 @@ MSBuild too._
+    <TargetFrameworks>netstandard2.0;netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
+
+  <PropertyGroup TargetFramework="netstandard">
+    <SomeProperty>Some value that applies to all versions of .NET Standard<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup TargetFramework=">=netcoreapp2.0">
+    <SomeProperty>Some value that applies to .NET Core 2.0 and later<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup TargetFramework="==net5.0-ios13.0">
+    <SomeProperty>Some value that only applies to .NET 5 + iOS 13.0<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup TargetPlatform="windows">
+    <SomeProperty>Some value that applies to all version of Windows<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup TargetPlatform=">=ios-12.0">
+    <SomeProperty>Some value that applies to iOS 12.0 and later<SomeProperty>
+  </PropertyGroup>
+
+</Project>
+```
+
+The idea is:
+
+* Property groups, properties, and item groups get new attributes
+  `TargetFramework` and `TargetPlatform`.
+* The value can be prefixed with an optional conditional operator `==`, `!=`,
+  `<`, `<=`, `>`, and `>=`. If the operator is omitted, `==` is assumed.
+* `TargetFramework` supports comparisons with a friendly TFM name. This can
+  include an OS flavor for symmetry. If the `TargetFramework` property includes
+  an OS flavor but the attribute doesn't, the comparison only applies to the TFM
+  without the OS flavor. In other words a condition of
+  `TargetFramework=">=net5.0"` will result in `true` if the project targets
+  `net5.0`, `net6.0`, as well as `net6.0-android12.0`.
+
+Alternatively, we could invent new syntax that allows parsing of constitutes
+like this:
+
+```XML
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(TargetFramework::Identifier)=='netstandard'">
+    <SomeProperty>Some value that applies to all versions of .NET Standard<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(TargetFramework::Name)>='netcoreapp2.0'">
+    <SomeProperty>Some value that applies to .NET Core 2.0 and later<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(TargetFramework::Name)=='net5.0-ios13.0'">
+    <SomeProperty>Some value that only applies to .NET 5 + iOS 13.0<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(TargetFramework::Platform)=='windows'">
+    <SomeProperty>Some value that applies to all version of Windows<SomeProperty>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(TargetFramework::Platform)>='ios-12.0'">
+    <SomeProperty>Some value that applies to iOS 12.0 and later<SomeProperty>
+  </PropertyGroup>
+
+</Project>
+```
+
+And lastly, we could just define new intrinsic functions on some type,
+but this will make using them a mouthful:
+
+```XML
+  <PropertyGroup Condition="`'$([MSBuild]::TargetFrameworkIdentifier($(TargetFramework)))' == '.NETStandard'`">
```

Comment:
This is the most "like the rest of existing MSBuild". Of course that means it's super ugly :)

Another option would be to create a new [intrinsic function for conditions](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-conditions?view=vs-2019) (we currently have only `Exists` and `HasTrailingSlash`).

Reply:
Moved to: https://github.com/microsoft/msbuild/issues/5171